hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/package-info.java

```diff
@@ -72,12 +72,25 @@
  * hashtables (hence the name). It consumes the enriching streams in full
  * before ingesting any data from the primary stream.
  * <p>
- * The output is just like an SQL left-outer join: for each primary item
- * there are N output items, one for each matching item in the enriching
- * set. If an enriching set doesn't have a matching item, the output will
- * have a {@code null} instead of the enriching item. The join also allows
+ * The output of {@code hashJoin} is just like an SQL left-outer join:
+ * for each primary item there are N output items,
+ * one for each matching item in the enriching set.
+ * If an enriching set doesn't have a matching item, the output will
+ * have a {@code null} instead of the enriching item.
+ * <p>
+ * If SQL inner-join is needed,
```

Comment:
Please try to use the active voice as much as possible. This can be "If you need the inner join, ". Also note "inner join" (without a dash). We actually need to fix "hash-join" into "hash join" as well, but this is not the topic of this PR.